services/QuillLMS/app/lib/rule_feedback_history.rb

```diff
@@ -45,10 +48,13 @@ def self.feedback_history_to_json(f_h)
         }
     end
 
-    def self.generate_rulewise_report(rule_uid:, prompt_id:, start_date: nil, end_date: nil)
+    def self.generate_rulewise_report(rule_uid:, prompt_id:, start_date: nil, end_date: nil, turk_session_id: nil)
         feedback_histories = FeedbackHistory.where(rule_uid: rule_uid, prompt_id: prompt_id, used: true)
-        feedback_histories = feedback_histories.where("created_at >= ?", start_date) if start_date
-        feedback_histories = feedback_histories.where("created_at <= ?", end_date) if end_date
+        feedback_histories = feedback_histories.joins('LEFT JOIN feedback_sessions ON feedback_histories.feedback_session_uid = feedback_sessions.uid')
+        feedback_histories = feedback_histories.joins('LEFT JOIN comprehension_turking_round_activity_sessions ON feedback_sessions.activity_session_uid = comprehension_turking_round_activity_sessions.activity_session_uid')
+        feedback_histories = feedback_histories.where("feedback_histories.created_at >= ?", start_date) if start_date
+        feedback_histories = feedback_histories.where("feedback_histories.created_at <= ?", end_date) if end_date
+        feedback_histories = feedback_histories.where("comprehension_turking_round_activity_sessions.turking_round_id = ?", turk_session_id) if turk_session_id
```

Comment:
Same as above.  You could put the joins inside the same condition that you use for your `if turk_session_id` `WHERE` clause if you wanted.